classes/Hook.php

```diff
@@ -916,7 +954,54 @@ public static function exec(
             $hookRegistry->collect();
         }
 
-        return $output;
+        if ($array_return) {
+            return $output;
+        }
+        return ($liveEdit ? '<div id="'.$hook_name.'" class="dndHook" style="min-height:50px">' : '').$output.($liveEdit ? '</div>' : '');
+        // Return html string
+    }
+
+    /**
+     * Regenerate content of hook by adding html identifier
+     *
+     * @param string $display Get content for this hook
+     * @param string $moduleInstance Get modules content
+     * @param int $hookId Get hook id
+     * @return string
+     */
+    public static function wrapLiveEdit($display, $moduleInstance, $hookId)
+    {
+
+        $context = Context::getContext();
+
+        $linkModule = sprintf(
+            '%s%s/index.php?controller=AdminModules&configure=%s&token=%s',
+            $context->link->getBaseLink(),
+            Tools::getValue('ad'),
+            Tools::safeOutput($moduleInstance->name),
+            Tools::getAdminToken('AdminModules'.(int)Tab::getIdFromClassName('AdminModules').(int)Tools::getValue('id_employee'))
+        );
+
+        $isConfigurable = method_exists($moduleInstance, 'getContent') ? 1 : 0;
+
+        $liveContent = '<div rel="'.Tools::safeOutput($moduleInstance->name).'" id="hook_'.(int)$hookId.'_module_'.(int)$moduleInstance->id.'_moduleName_'.str_replace('_', '-', Tools::safeOutput($moduleInstance->name)).'"
```

Comment:
Please create a new tpl dedicated to wrapping `$display` wih live edit html. Don't do it directly in this method.